### PR TITLE
Add io.gitlab.azymohliad.WatchMate

### PR DESCRIPTION
**Pull request by azymohliad**
### Please confirm your submission meets all the criteria

- [x] I have read the [App Requirements][reqs] and [App Maintenance][maint] pages.
- [x] My pull request follows the instructions at [App Submission][submission].
- [x] I am using only the minimal set of permissions. *(If not, please explain each non-standard permission.)*
- [x] All assets referenced in the manifest are redistributable by any party.  If not, the unredistributable parts are using an extra-data source type.
- [x] I am an upstream contributor to the project. If not, I contacted upstream developers about submitting their software to Flathub. **Link:**
- [x] I own the domain used in the application ID or the domain has a policy for delegating subdomains (e.g. GitHub, SourceForge).
- [x] Any additional patches or files have been submitted to the upstream projects concerned. *(If not, explain why.)*

[reqs]: https://github.com/flathub/flathub/wiki/App-Requirements
[maint]: https://github.com/flathub/flathub/wiki/App-Maintenance
[submission]: https://github.com/flathub/flathub/wiki/App-Submission

### CODE REVIEW

File: io.gitlab.azymohliad.WatchMate.yml [hunk 1]
```diff
@@ -0,0 +1,37 @@
...
+  - --share=ipc
+  - --share=network
+  - --socket=wayland
+  - --socket=fallback-x11
+  - --system-talk-name=org.bluez
+  - --talk-name=org.mpris.MediaPlayer2.*
+  - --filesystem=xdg-download
+modules:
+  - name: watchmate
+    sources:
```
Comment (hfiguiere) [inline io.gitlab.azymohliad.WatchMate.yml:19]: can we make `sources` the last item here by convention?

File: io.gitlab.azymohliad.WatchMate.yml [hunk 2]
```diff
@@ -0,0 +1,37 @@
...
+      env:
+        CARGO_HOME: /run/build/watchmate/cargo
+    build-commands:
+      - cargo --offline fetch --manifest-path Cargo.toml --verbose
+      - cargo --offline build --release --verbose
+      - install -Dm 755 target/release/watchmate -t /app/bin
+      - install -Dm 644 assets/${FLATPAK_ID}.desktop -t /app/share/applications
+      - install -Dm 644 assets/${FLATPAK_ID}.metainfo.xml -t /app/share/metainfo
+      - install -Dm 644 assets/icons/${FLATPAK_ID}.svg -t /app/share/icons/hicolor/scalable/apps
+      - install -Dm 644 assets/icons/${FLATPAK_ID}-symbolic.svg -t /app/share/icons/hicolor/symbolic/app
```
Comment (hfiguiere) [inline io.gitlab.azymohliad.WatchMate.yml:37]: I think you meant this:
```suggestion
      - install -Dm 644 assets/icons/${FLATPAK_ID}-symbolic.svg -t /app/share/icons/hicolor/scalable/apps
```

**Comment by hfiguiere** [comment]
bot, build io.gitlab.azymohliad.WatchMate

File: io.gitlab.azymohliad.WatchMate.yml (hunk 2 above)
Comment (azymohliad) [reply io.gitlab.azymohliad.WatchMate.yml:37]: Oh, is that for sure? I can't find any documentation quickly, but I see [gtk-rust-template uses `symbolic/apps`](https://gitlab.gnome.org/World/Rust/gtk-rust-template/-/blob/master/data/icons/meson.build) for symbolic icon. I have it wrong anyway tho, at least need to change `app` to `apps`

Comment (azymohliad) [reply io.gitlab.azymohliad.WatchMate.yml:37]: It seems all other flatpak apps I have installed use `symbolic/apps` for symbolic icons, so I'll only fix `app` -> `apps` here for now. But let me know if it's actually incorrect, and I'll change to `scalable`.

```
$ find /var/lib/flatpak/app/**/apps -name "*-symbolic.svg" | grep symbolic | wc -l
240
$ find /var/lib/flatpak/app/**/apps -name "*-symbolic.svg" | grep scalable | wc -l
0
```

**Comment by hfiguiere** [comment]
```
builddir/files/share/appdata/io.gitlab.azymohliad.WatchMate.appdata.xml: FAILED:
• style-invalid         : Content before <ul> is too short [8], at least 20 characters required
Validation of files failed
```

**Comment by azymohliad** [comment]
Yeah, I'm fixing it... It seems a bit of a false positive for the verification, it complains about [this line](https://gitlab.com/azymohliad/watchmate/-/blob/dc153113d6f31769e59dcbb238f5828fbd4f3a8b/assets/io.gitlab.azymohliad.WatchMate.metainfo.xml#L24), but it would be really unnatural to put anything between features list and the roadmap. I'll probably just remove the roadmap

**Comment by azymohliad** [comment]
I changed the commit to which the tag `v0.2.0` points a couple of times. I hope it's ok before the app is published?

**Comment by azymohliad** [comment]
bot, build io.gitlab.azymohliad.WatchMate

**Comment by azymohliad** [comment]
bot, build io.gitlab.azymohliad.WatchMate

**Comment by azymohliad** [comment]
bot, build io.gitlab.azymohliad.WatchMate

**Comment by azymohliad** [comment]
Something definitely went wrong with the CI build, it's stuck for 1 hour. @hfiguiere, sorry for the ping, I guess it's better to cancel it to avoid wasting CI time.

**Comment by barthalion** [comment]
bot, build io.gitlab.azymohliad.WatchMate

**Comment by azymohliad** [comment]
The reason for the last CI failures is unclear to me :/ The latest x86_64 build succeeded without me changing anything, while the one before that failed with timeout (stdout is silent for 1h). The latest aarch64 failed as well due to this timeout. Seems to be stuck while downloading the dependencies. Locally flatpak-builder works. Not sure if I should just retry...

**Comment by azymohliad** [comment]
bot, build io.gitlab.azymohliad.WatchMate

**Comment by barthalion** [comment]
bot, build io.gitlab.azymohliad.WatchMate

**Pull request closed without merging**